src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMParams.scala

```diff
@@ -325,4 +325,11 @@ trait LightGBMParams extends Wrappable with DefaultParamsWritable with HasWeight
 
   def getMinDataInLeaf: Int = $(minDataInLeaf)
   def setMinDataInLeaf(value: Int): this.type = set(minDataInLeaf, value)
+
+  var delegate: Option[LightGBMDelegate] = None
```

Comment:
we should probably figure out how to make this inherit from param and expose in python layer, but maybe not for this PR